lib/ASRouter.jsm

```diff
@@ -435,7 +436,12 @@ class _ASRouter {
       message = await this._findMessage(msgs, target, data);
     }
 
-    await this.setState({lastMessageId: message ? message.id : null});
+    if (previewMsgs.length) {
+      // We don't want to cache preview messages, remove them after we selected the message to show
+      await this.setState(state => this._removePreviewMessage(state, {lastMessageId: message.id}));
```

Comment:
are we sure there will be a `message` ? the else case seems to guard against message not existing, should we do that here too?

Reply:
if `previewMsgs.length` then we do have a message. We pick the first element of the array to dispatch.